### DISCUSSION TITLE
Chart dependency error after Angular 17 migration with new esbuild builder

### DISCUSSION THREAD

**Discussion by ips219** [primefaces/community#1838] [PrimeNG]
I upgraded previously  to Angular 17 with webpack builder and module federation plugin and the application worked correctly. In this second migration step, I am changing the builder to the new esbuild based build system, and I'm having some issues with chart component

`
main.ts:8 Error: Unable to resolve specifier 'chart.js/auto' imported from http://localhost:4200/primeng_chart-17_15_0-dev.js
`

It seems from the debugger information, it has a problem importing from chart.js/auto

`
import * as i0 from "@angular/core";
import { EventEmitter, PLATFORM_ID, booleanAttribute, Component, ChangeDetectionStrategy, ViewEncapsulation, Inject, Input, Output, NgModule } from "@angular/core";
import { isPlatformBrowser, CommonModule } from "@angular/common";
import Chart from "chart.js/auto";
`

In the package.json I have chart.js:4.4.2 dependency included.

In my previous Angular.json configuration file, I had this setting

`
					"allowedCommonJsDependencies": [
						"chart.js"
					],
`

which is no longer supported in the new builder (I removed it), and in the scripts property I also had the reference to the chartjs library

`
						"scripts": [
							"node_modules/chart.js/dist/chart.js"
						],
`

which makes the build process fail with this error:

`
Failed to resolve dependency: ./chunks/helpers.segment.js, present in 'optimizeDeps.include'
Failed to resolve dependency: @kurkle/color, present in 'optimizeDeps.include'
`

I've tried to change the script to the umd bundle which builds correctly

`
						"scripts": [
							"node_modules/chart.js/dist/chart.umd.js"
						],
`

But then the application fails to start-up aruntime with the given error.

Did anybody faced a similar issue after migrating to esbuild?

Best regards.

**Comment by cetincakiroglu** [comment]
Hi,

You're using chartjs 4.4.2, PrimeNG uses chartjs 3.3.2. The issue probably raised from the version difference, could you please try it with the 3.3.2 until we upgrade the chart.js version?

**Reply by ips219** [reply]
Hi,

Thanks for your response @cetincakiroglu, after downgrading to 3.3.2 that error seems to be gone.

It is strange as I was working with 4.3.0 of ChartJs with Angular 16 before starting the migration (I don't remember exactly why we migrated, I think it was required by one chartjs plugin) without issues...

NowI have some other details related with esbuild migration but it seems that they are not related with primeng.

Thanks for your support!
